src/py/flwr/common/config.py

```diff
@@ -113,8 +113,13 @@ def get_fused_config(run: Run, flwr_dir: Optional[Path]) -> UserConfig:
     return get_fused_config_from_dir(project_dir, run.override_config)
 
 
-def flatten_dict(raw_dict: Dict[str, Any], parent_key: str = "") -> UserConfig:
+def flatten_dict(
+    raw_dict: Optional[Dict[str, Any]], parent_key: str = ""
```

Comment:
```suggestion
    raw_dict: Optional[UserConfig], parent_key: str = ""
```

Reply:
@danieljanes No, this function needs to receive a raw dict of type `dict[str, Any]` as it can be a `dict` of `dict`